src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1140,13 +1129,7 @@ _mongoc_client_new_from_uri (mongoc_topology_t *topology)
    const char *appname;
 
    BSON_ASSERT (topology);
-
-#ifndef MONGOC_ENABLE_SSL
-   if (mongoc_uri_get_tls (topology->uri)) {
-      MONGOC_ERROR ("Can't create SSL client, SSL not enabled in this build.");
-      return NULL;
-   }
-#endif
+   BSON_ASSERT (topology->valid);
```

Comment:
After @eramongodb suggested some assertions on this function's return value, I realized that the code is always checking `topology->valid` before invoking `_mongoc_client_new_from_topology`. There should be no case where we call this with an invalid topology, and asserting a non-null return value is equivalent to asserting the one condition that would have resulted in such a return value.

It seemed simpler to just remove the error condition and assert directly. I don't see the harm in retaining asserts on the returned `client` pointer, but I'm open to removing those if you like.

@eramongodb: This change also avoids any need to document ownership of the `topology` on failure, since this function will no longer fail (short of a `SIGABRT`),

Reply:
@jmikola Sweet. I think it would then be appropriate to add a comment/documentation to `_mongoc_client_new_from_topology()` that the provided topology must be valid as a precondition.

Reply:
Precondition is now documented.

Another point to mention: whether or not `client` ends up _owning_ the `topology` still depends on whether the client is single-threaded (see: `mongoc_client_destroy`). So I'm glad we don't have to add any comment about `topology` ownership.